.github/workflows/ci-tests.yml

```diff
@@ -184,3 +184,43 @@ jobs:
       - uses: actions/checkout@v3
       - name: build & test cwltool_module container
         run: ./build-cwltool-docker.sh
+
+  macos:
+    name: CI test on macos-latest
+    runs-on: macos-latest
+    env:
+      TOXENV: py310-unit
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          cache: pip
+          cache-dependency-path: |
+            requirements.txt
+            tox.ini
+      - name: Upgrade setuptools and install tox
+        run: |
+          pip install -U pip setuptools wheel
+          pip install tox tox-gh-actions
+      # # docker for mac install is not currently stable
```

Comment:
I coudn't find an issue in their repository. What's the problem with the action? Is it, perhaps, something with their latest commits? Maybe we could pin to a previous commit or one of their tags?

I tested uncommenting these lines, cherry-picking the #1715 commit, and got 6 failures. One similar to what @jfennick reported, with the failure at `assert 5 == 2`.

Here's my GH Action macos job: https://github.com/kinow/cwltool/runs/7911944514

Reply:
@kinow It was very inconsistent for me, and I saw others comments confirming this.